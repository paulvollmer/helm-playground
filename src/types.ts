declare global {
  interface Window {
    helmRender: (templates: string, values: string, chart: string, settings: string) => helmRenderReturn
    helmDefaultCapabilities: () => SettingsData
  }
}

export type helmRenderReturn = {
  error?: helmRenderReturnError
  result?: string
}

export type helmRenderReturnError = {
  kind: string
  file: string
  line: number
  message: string
}

export type Sources = {
  [key: string]: string
}

export type SettingsData = {
  release: SettingsRelease
  kubeVersion: SettingsKubeVersion
  helmVersion: SettingsHelmVersion
}

export type SettingsRelease = {
  name: string
  namespace: string
  revision: string
  isUpgrade: string
  isInstall: string
  service: string
}

export type SettingsKubeVersion = {
  version: string
  major: string
  minor: string
}

export type SettingsHelmVersion = {
  version: string
  gitCommit: string
  gitTreeState: string
  goVersion: string
}
