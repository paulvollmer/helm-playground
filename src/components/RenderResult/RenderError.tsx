import React from 'react'
import { Typography } from '@mui/material'
import { HelmRenderReturnError } from '../../types'

export type RenderErrorProps = {
  error: HelmRenderReturnError
}

export const RenderError = (props: RenderErrorProps): JSX.Element => {
  return (
    <div>
      <Typography variant="subtitle1">Error Kind: {props.error.kind}</Typography>
      <Typography variant="h6">{props.error.message}</Typography>
      <Typography variant="body1">
        File: {props.error.file}
        <br />
        {props.error.line ? `Line: ${props.error.line}` : ''}
      </Typography>
    </div>
  )
}
