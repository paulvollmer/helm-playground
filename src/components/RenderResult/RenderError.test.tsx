import React from 'react'
import TestRenderer from 'react-test-renderer'
import { RenderError } from './RenderError'
import { HelmRenderReturnError } from '../../types'

test('renders RenderError', () => {
  const testError: HelmRenderReturnError = {
    kind: 'test-kind',
    file: 'test-file',
    line: 1,
    message: 'test-message',
  }
  const tree = TestRenderer.create(<RenderError error={testError} />).toJSON()
  expect(tree).toMatchSnapshot()
})
