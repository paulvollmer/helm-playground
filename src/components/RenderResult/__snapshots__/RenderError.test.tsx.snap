// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RenderError 1`] = `
<div>
  <h6
    className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
  >
    Error Kind: 
    test-kind
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
  >
    test-message
  </h6>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    File: 
    test-file
    <br />
    Line: 1
  </p>
</div>
`;
